app/models/guide_author.ts

```diff
@@ -0,0 +1,21 @@
+import { BaseModel, column, manyToMany } from "@adonisjs/lucid/orm";
+import * as relations from "@adonisjs/lucid/types/relations";
+
+import GuideArticle from "#models/guide_article";
+
+export default class GuideAuthor extends BaseModel {
+  @column({ isPrimary: true })
+  declare id: number;
+
+  @column()
+  declare name: string;
+
+  @manyToMany(() => GuideArticle, {
+    pivotTable: "guide_article_authors",
+    pivotColumns: ["role"],
+    pivotForeignKey: "author_id",
+    pivotRelatedForeignKey: "article_id",
+    relatedKey: "id",
+  })
+  declare guideArticles: relations.ManyToMany<typeof GuideArticle>;
```

Comment:
might want to enable `pivotTimestamps` on this relation
```suggestion
  @manyToMany(() => GuideAuthor, {
    pivotTable: "guide_article_authors",
    pivotColumns: ["role"],
    pivotForeignKey: "article_id",
    pivotRelatedForeignKey: "author_id",
    relatedKey: "id",
    pivotTimestamps: true,
  })
  declare guideAuthors: relations.ManyToMany<typeof GuideAuthor>;
```